app/crawler/naver_map_crawler.py

```
@@ -147,35 +160,100 @@ def _search_sync(self, query: str) -> List[Dict[str, str]]:
+        return max(0, int(base_ms + random.randint(0, self.PAGE_WAIT_JITTER_MS)))
+
+    def _goto_search_page_with_retry(self, page, url: str) -> bool:
+        max_attempts = self.RATE_LIMIT_RETRIES + 1
+
+        for attempt in range(max_attempts):
+            response = page.goto(url)
+            status = getattr(response, "status", None)
+
+            if status == 429:
+                backoff = self.RATE_LIMIT_BACKOFF_SEC * (2**attempt)
+                jitter = random.uniform(0, max(self.RATE_LIMIT_BACKOFF_SEC, 0.0))
```

Ruff (0.15.2)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)